beacon-chain/rpc/eth/shared/request.go

```diff
@@ -134,3 +137,36 @@ func IsOptimistic(
 	http2.WriteError(w, errJson)
 	return true, nil
 }
+
+// DecodeWithLength takes a string and a length and validates the hex while returning an error
+func DecodeWithLength(s string, length int) ([]byte, error) {
+	bytes, err := hexutil.Decode(s)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("%s is invalid", s))
+	}
+	if len(bytes) != length {
+		return nil, fmt.Errorf(" %s is not length %d", s, length)
```

Comment:
```suggestion
		return nil, fmt.Errorf("%s is not length %d bytes", s, length)
```
Without saying `bytes` it sounds like you are checking the length of the string, which will be confusing